graylog2-server/src/main/java/org/graylog2/migrations/V20170607164210_MigrateReopenedIndicesToAliases.java

```diff
@@ -0,0 +1,138 @@
+    private static final String REOPENED_INDEX_SETTING = "graylog2_reopened";
+
+    private Node node;
+    private final IndexSetService indexSetService;
+    private final MongoIndexSet.Factory mongoIndexSetFactory;
+    private final Indices indices;
+    private final JestClient jestClient;
+
+    @Inject
+    public V20170607164210_MigrateReopenedIndicesToAliases(Node node,
+                                                           IndexSetService indexSetService,
+                                                           MongoIndexSet.Factory mongoIndexSetFactory,
+                                                           Indices indices,
+                                                           JestClient jestClient) {
+        this.node = node;
+        this.indexSetService = indexSetService;
+        this.mongoIndexSetFactory = mongoIndexSetFactory;
+        this.indices = indices;
+        this.jestClient = jestClient;
+    }
+
+    @Override
+    public ZonedDateTime createdAt() {
+        return ZonedDateTime.parse("2017-06-07T16:42:10Z");
+    }
+
+    // Create aliases for legacy reopened indices.
+    @Override
+    public void upgrade() {
+        this.indexSetService.findAll()
+            .stream()
+            .map(mongoIndexSetFactory::create)
+            .flatMap(indexSet -> getReopenedIndices(indexSet).stream())
+            .map(indexName -> { LOG.debug("Marking index {} to be reopened using alias.", indexName); return indexName; })
+            .forEach(indices::markIndexReopened);
+    }
+
+    private Set<String> getReopenedIndices(final Collection<String> indices) {
+        final Version elasticsearchVersion = node.getVersion().orElseThrow(() -> new ElasticsearchException("Unable to retrieve Elasticsearch version."));
+        final String indexList = String.join(",", indices);
+        final State request = new State.Builder().withMetadata().indices(indexList).build();
+
+        final JestResult jestResult = JestUtils.execute(jestClient, request, () -> "Couldn't read cluster state for reopened indices " + indices);
+        final JsonNode clusterStateJson = jestResult.getJsonObject();
+        final JsonNode indicesJson = clusterStateJson.path("metadata").path("indices");
+
+        final ImmutableSet.Builder<String> reopenedIndices = ImmutableSet.builder();
+
+        if (indicesJson.isMissingNode()) {
+            LOG.error("Retrieved cluster state is invalid (no metadata.indices key).");
+            LOG.debug("Received cluster state was: {}", clusterStateJson.toString());
+            return Collections.emptySet();
+        }
+
+        for (Iterator<Map.Entry<String, JsonNode>> it = indicesJson.fields(); it.hasNext(); ) {
+            final Map.Entry<String, JsonNode> entry = it.next();
+            final String indexName = entry.getKey();
+            final JsonNode value = entry.getValue();
+
+            final JsonNode indexSettings = value.path("settings").path("index");
```

Comment:
Here we don't catch the `archived` section of the index settings in case we are upgrading.
I think we need to pass in the entire `value` into `checkForReopened` or simply inline that code and perform the missing settings check after the version check.

This happens when upgrading from 2.4.2 to 5.4.2 while migrating the reopen check.